causalml/inference/tree/uplift.pyx

```diff
@@ -224,18 +224,29 @@ class UpliftTreeClassifier:
         The normalization factor defined in Rzepakowski et al. 2012, correcting for tests with large number of splits
         and imbalanced treatment and control splits.
 
+    honesty: bool (default=False)
+         True if the honest approach based on "Athey, S., & Imbens, G. (2016). Recursive partitioning for heterogeneous causal effects."
+         shall be used. If 'IDDP' is used as evaluation function, this parameter is automatically set to true.
+
+    estimation_sample_size: float (default=0.5)
+         Sample size for estimating the CATE score in the leaves if honesty == True.
+
     random_state: int, RandomState instance or None (default=None)
         A random seed or `np.random.RandomState` to control randomness in building a tree.
 
     """
     def __init__(self, control_name, max_features=None, max_depth=3, min_samples_leaf=100,
                  min_samples_treatment=10, n_reg=100, evaluationFunction='KL',
-                 normalization=True, random_state=None):
+                 normalization=True, honesty=False, estimation_sample_size=0.5, random_state=None):
         self.max_depth = max_depth
         self.min_samples_leaf = min_samples_leaf
         self.min_samples_treatment = min_samples_treatment
         self.n_reg = n_reg
         self.max_features = max_features
+
+        assert evaluationFunction is not None and evaluationFunction in ['KL', 'ED', 'Chi', 'CTS', 'DDP', 'IT', 'CIT', 'IDDP'], \
```

Comment:
The first condition, `evaluationFunction is not None`, is not necessary because if `evaluationFunction` is `None`, it won't meet the second condition anyway.